File: src/main/java/pl/yalgrin/playnite/simplesync/web/objects/AgeRatingResource.java
package pl.yalgrin.playnite.simplesync.web.objects;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.objects.AgeRatingDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.objects.AgeRatingService;
import reactor.core.publisher.Mono;

@RestController
@Slf4j
@RequestMapping("/api/age-rating")
@RequiredArgsConstructor
public class AgeRatingResource {
    private final AgeRatingService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<AgeRatingDTO> getAgeRating(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<AgeRatingDTO> saveAgeRating(@RequestBody AgeRatingDTO dto) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteAgeRating(@RequestBody AgeRatingDTO dto) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/AgeRatingResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.dto.objects.AgeRatingDTO
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.service.objects.AgeRatingService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/age-rating")
class AgeRatingResource(
    private val service: AgeRatingService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getAgeRating(@PathVariable id: Long): Mono<AgeRatingDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveAgeRating(@RequestBody dto: AgeRatingDTO): Mono<AgeRatingDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto))
    }

    @PostMapping("/delete")
    fun deleteAgeRating(@RequestBody dto: AgeRatingDTO): Mono<Void> {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto))
    }
}